x-pack/plugin/frozen-indices/src/main/java/org/elasticsearch/xpack/frozen/rest/action/RestFreezeIndexAction.java

```diff
@@ -22,11 +23,18 @@
 
 public final class RestFreezeIndexAction extends BaseRestHandler {
 
+    public static final String DEPRECATION_WARNING = "Frozen indices are deprecated because they provide no benefit given improvements " +
+        "in heap memory utilization. They will be removed in a future release.";
+    private static final RestApiVersion DEPRECATION_VERSION = RestApiVersion.V_8;
+
     @Override
     public List<Route> routes() {
         return List.of(
-            new Route(POST, "/{index}/_freeze"),
-            new Route(POST, "/{index}/_unfreeze"));
+            Route.builder(POST, "/{index}/_freeze")
+                .deprecated(DEPRECATION_WARNING, DEPRECATION_VERSION)
+                .build(),
+            Route.builder(POST, "/{index}/_unfreeze").build()
```

Comment:
We should also deprecate this endpoint, because it will also technically be going away. If we don't show the message a user could hit this a bunch but not know it's going away.

Reply:
I was operating off the deprecation schedule listed in the [meta issue](https://github.com/elastic/elasticsearch/issues/70192) in which `unfreeze` deprecation happens in 8.0. I'm open to deprecating it earlier though I'd like to get buy-in from at least some others involved in the meta issue.

Reply:
@zuketo, could you weigh in on the question of deprecating the `unfreeze` endpoint now versus in 8.0?